pandas/types/inference.py

```diff
@@ -52,21 +233,86 @@ def is_re_compilable(obj):
         return True
 
 
-def is_list_like(arg):
-    return (hasattr(arg, '__iter__') and
-            not isinstance(arg, string_and_binary_types))
+def is_list_like(obj):
+    """
+    Check if the object is list-like.
```

Comment:
Maybe give an inexhaustive list of objects that are regarded as list-like? (lists, sets, arrays, Series, ...)

I would also add example with np.array(..)